pkg/occlient/occlient.go

```diff
@@ -2076,6 +2076,63 @@ type Service struct {
 	PlanList []string
 }
 
+// GetServiceClass retrieves the ClusterServiceClass identified by the specified external name
+func (c *Client) GetServiceClass(externalName string) (class *scv1beta1.ClusterServiceClass, err error) {
+	classes, err := c.serviceCatalogClient.ClusterServiceClasses().List(metav1.ListOptions{
+		FieldSelector: "spec.externalName==" + externalName,
+	})
+
+	if len(classes.Items) != 1 {
+		return nil, fmt.Errorf("%s: unknown service class", externalName)
+	}
+
+	return &classes.Items[0], err
+}
+
+// GetServiceClassesByCategory retrieves a map associating category name to ClusterServiceClasses matching the category
+func (c *Client) GetServiceClassesByCategory() (categories map[string][]scv1beta1.ClusterServiceClass, err error) {
+	categories = make(map[string][]scv1beta1.ClusterServiceClass)
+	classes, err := c.GetClusterServiceClasses()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get cluster service classes")
+	}
+
+	// TODO: Should we replicate the classification performed in
+	// https://github.com/openshift/console/blob/master/frontend/public/components/catalog/catalog-items.jsx?
+	for _, class := range classes {
+		tags := class.Spec.Tags
+		category := "other"
+		if len(tags) > 0 {
+			category = tags[0]
+		}
+		if len(category) > 0 {
```

Comment:
Isn't this condition always true? I mean `category:="other"` is just above it always

Reply:
Because maybe `category = tags[0]` is an empty string.

Reply:
But if `category = tags[0]` is empty, then category will be "" with length 0 and thus the class won't be assigned to the `other` category. Or am I missing something?

Reply:
Can you use the empty string `""` as a key in a go map? If yes, then you need this check (because we don't want that to happen in this case). If not, then you're right, we don't need this check. At the very least, it doesn't hurt :)

Reply:
No I meant if len(tags)>0 && tags[0] is "' then 
```
if len(tags) > 0 {
			category = tags[0]
		}
```
After this condition, category will become "" and the check of `if len(category) > 0 {}` will be false and thus the class will not be appended to the map

Reply:
@metacosm Still waiting for your response on this :)

Reply:
@mik-dass I missed it, sorry. Should be fixed now. Thank you!